### PR TITLE
Detect whether USB data pins are connected

### PR DESCRIPTION
**Pull request by hiviah**
Implementation of  #2340.

### CODE REVIEW

File: core/embed/unix/usb.c [hunk 1]
```diff
@@ -249,3 +250,11 @@ int usb_webusb_write(uint8_t iface_num, const uint8_t *buf, uint32_t len) {
 void pendsv_kbd_intr(void) {}
 
 void mp_hal_set_vcp_iface(int iface_num) {}
+
+secbool usb_configured(void) {
+  if (access("/var/tmp/trezor.usb_data_disconnected", F_OK) == 0) {
```
Comment (matejcik) [inline core/embed/unix/usb.c:255]: please add to `unix/profile.c` instead of hard-coding the name

File: core/src/apps/homescreen/homescreen.py [hunk 2]
```diff
@@ -54,6 +67,9 @@ def do_render(self) -> None:
         # homescreen with shifted avatar and text on bottom
         # Differs for each model
 
+        if not utils.usb_data_connected():
```
Comment (matejcik) [inline core/src/apps/homescreen/homescreen.py:70]: please render a header instead.
`ui.header_error("NO USB CONNECTION")` might fit?
this error is the highest priority so it should be the first check in the list

Comment (hiviah) [reply core/src/apps/homescreen/homescreen.py:70]: Changed.

**Review by matejcik — APPROVED** [review]
LGTM
will merge once the CI goes green

(next time please don't resolve comments which you didn't start, the reviewer should do that)

we should add a test for this, but it's not really possible currently; we would either need to add emulator control to device_tests, or introduce UI recording to persistence_tests. I will create a separate issue for that.

**Comment by hiviah** [comment]
Everything in CI is green now.

(don't understand why the click test got stuck before but everything looks OK now)

**Comment by matejcik** [comment]
QA instructions:
- get a data-only USB cable; or plug the TT into a charger instead of a PC
- check that "NO USB CONNECTION" is shown on the homescreen (only after unlocking with PIN, i believe)
- check that it is _not_ shown when connecting to a normal PC with a good cable

**Comment by matejcik** [comment]
@hiviah do we have any way for a normal user to trigger the _change_ in this parameter?
i dunno, get a powered USB hub, leave it unplugged from PC, then plug into PC?

**Comment by bosomt** [comment]
No idea what happened here.
I'm using 2.5.3 for some time but i seen it for first time.
I was running CoinJoin so device was not able to go to sleep.
As soon i added another account in Trezor Suite message was gone

image


**Info:**
 - Suite version: desktop 22.11.1 (36517f2367056ec69f25858ddd96684364d1dbc5)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/22.11.1 Chrome/100.0.4896.160 Electron/18.3.5 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: model T 2.5.3 Universal (revision 2f03ace311584988d5aeab58fd1acf24ef54711a)
 - Transport: bridge 2.0.32

**Comment by hiviah** [comment]
Does it happen also when you set long autolock to prevent locking and just leave it connected for the same amount of time, without doing anything? How long did it take before the message showed up?

My guess is that OS deconfigured&suspended the device and once you try to connect to it again (make an account), bridge triggered OS to reenumerate USB devices so the device went back to "configured" (actively connected state).

**Comment by bosomt** [comment]
@hiviah device/coinjoin was running all night long during Thursday night. So its possible it after autolock time. Found it like this at Friday morning.

**Comment by bosomt** [comment]
it just happened to me again

- Suite reported device connected
- bridge reported device connected and no session
- device reported `no usb conection`

as soon i added account in suite it changed to `pin not set`
so its definitely not PIN/autolock related, no PIN is set in device 

**Info:**
 - Suite version: desktop 22.12.0 (e97f817b93cf74d122ad303a2572a8fb1a9bc3f0)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/22.12.0 Chrome/104.0.5112.124 Electron/20.3.5 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: model T 2.5.3 Universal (revision 2f03ace311584988d5aeab58fd1acf24ef54711a)
 - Transport: bridge 2.0.32

**Pull request merged**